### ISSUE TITLE
Version updater does not support named tags

### ISSUE THREAD

**Issue by ivoleitao** [techouse/alfred_workflow#7]
Hi,

I have a mono repo with multiple packages, one of them a alfred workflow. Because of that I'm creating tags like this 'workflow_alfred-v2.3.0'. This is creating problems with the updater outputting errors like the following:

```
Unhandled exception:
FormatException: Could not parse "workflow_alfred-v2.3.2".
#0      new Version.parse (package:pub_semver/src/version.dart:120)
#1      VersionConverter.fromJson (package:alfred_workflow/src/converters/version_converter.dart:12)
#2      _$GithubReleaseFromJson (package:alfred_workflow/src/models/github_release.g.dart:298)
#3      new GithubRelease.fromJson (package:alfred_workflow/src/models/github_release.dart:69)
#4      AlfredUpdater.fetchLatestRelease (package:alfred_workflow/src/services/alfred_updater.dart:137)
<asynchronous suspension>
#5      AlfredUpdater.updateAvailable (package:alfred_workflow/src/services/alfred_updater.dart:89)
```

I've pinpointed this to the VersionConverter which mandates the tags to start with 'v' which is not always true as illustrated by this scenario. Is this something that could be changed ? For example by allowing the configuration of the extraction pattern or simply by regex matching

**Comment by techouse** [comment]
Hmm, I can have a look.

**Comment by ivoleitao** [comment]
> I've always stuck with `v1.2.3` because it makes the most sense and is easiest to parse.
> 
> Currently, it will parse a tag like `1.2.3` or `v1.2.3`, but you're correct to assume that `lorem_ipsum-v1.2.3` won't work.
> 
> ```dart
> @override
> Version fromJson(String string) => string.toLowerCase().startsWith('v')
>     ? Version.parse(string.substring(1))
>     : Version.parse(string);
> ```
> 
> Sp what you're essentially asking is for the version converter to allow for some custom regex which ignores anything before the numeric versions themselves.
> 
> I could expose that as some method that you can override.
> 
> Lemme see what I can do.

Yes exactly, some way to configure a specific pattern to allow tags like workflow_alfred-v2.3.0 to work

**Comment by techouse** [comment]
I just solved it with a regex. As long as the versioning uses semver `$major.$minor.$patch` it should work.

**Comment by techouse** [comment]
Fix released in `v0.2.10`

**Comment by ivoleitao** [comment]
Thank you, all working now
